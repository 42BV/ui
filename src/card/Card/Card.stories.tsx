import { storiesOf } from '@storybook/react';
import { CardTitle } from 'reactstrap';

import { Card } from './Card';

storiesOf('core/Card', module)
  .addParameters({ component: Card })
  .add('basic', () => {
    return (
      <Card
        header={<CardTitle>This is a card</CardTitle>}
        footer="And a footer"
      >
        <p>With some content in it</p>
      </Card>
    );
  })
  .add('without header', () => {
    return (
      <Card footer="And a footer">
        <CardTitle>This is a card</CardTitle>
        <p>With some content in it</p>
      </Card>
    );
  })
  .add('without footer', () => {
    return (
      <Card header={<CardTitle>This is a card</CardTitle>}>
        <p>With some content in it</p>
      </Card>
    );
  });
