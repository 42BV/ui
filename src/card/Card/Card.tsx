import { ReactNode, Suspense } from 'react';
import {
  Card as ReactstrapCard,
  CardBody,
  CardFooter,
  CardHeader,
  CardProps
} from 'reactstrap';
import { Loading } from '../../core/Loading/Loading';
import { ErrorBoundary } from 'react-error-boundary';
import { ContentState } from '../../core/ContentState/ContentState';
import { t } from '../../utilities/translation/translation';

export type Props = CardProps & {
  /**
   * Optionally the content of the card header.
   */
  header?: ReactNode;

  /**
   * Optionally the content of the card footer.
   */
  footer?: ReactNode;

  /**
   * Optional extra CSS class you want to add to the <CardHeader>.
   * Useful for styling the component.
   */
  cardHeaderClassName?: string;

  /**
   * Optional extra CSS class you want to add to the <CardBody>.
   * Useful for styling the component.
   */
  cardBodyClassName?: string;

  /**
   * Optional extra CSS class you want to add to the <CardFooter>.
   * Useful for styling the component.
   */
  cardFooterClassName?: string;
};

/**
 * Card is a content container with an optional header and footer.
 * You should use this component over the reactstrap Card component,
 * because this component has Suspense built in to display a loader
 * and wait for its content to be fully loaded before displaying.
 */
export function Card({
  header,
  footer,
  children,
  cardHeaderClassName,
  cardBodyClassName,
  cardFooterClassName,
  ...cardProps
}: Readonly<Props>) {
  return (
    <ReactstrapCard {...cardProps}>
      {header ? (
        <CardHeader className={cardHeaderClassName}>
          <Suspense fallback={<Loading />}>{header}</Suspense>
        </CardHeader>
      ) : null}
      <CardBody className={cardBodyClassName}>
        <Suspense fallback={<Loading />}>
          <ErrorBoundary
            fallback={
              <ContentState
                mode="error"
                title={t({
                  key: 'Card.ERROR',
                  fallback: 'Oops, something went wrong!'
                })}
              />
            }
          >
            {children}
          </ErrorBoundary>
        </Suspense>
      </CardBody>
      {footer ? (
        <CardFooter className={cardFooterClassName}>
          <Suspense fallback={<Loading />}>{footer}</Suspense>
        </CardFooter>
      ) : null}
    </ReactstrapCard>
  );
}
