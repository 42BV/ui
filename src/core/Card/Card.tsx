import React, { Suspense } from 'react';
import { Card as ReactstrapCard, CardBody, CardFooter, CardHeader } from 'reactstrap';
import { Loading } from '../Loading/Loading';
import { ErrorBoundary } from '../ErrorBoundary/ErrorBoundary';

export type Props = {
  /**
   * Optionally the content of the card header.
   */
  header?: React.ReactNode;

  /**
   * Optionally the content of the card footer.
   */
  footer?: React.ReactNode;

  /**
   * The content of the card body.
   */
  children: React.ReactNode;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;

  /**
   * Optional extra CSS class you want to add to the <CardHeader>.
   * Useful for styling the component.
   */
  cardHeaderClassName?: string;

  /**
   * Optional extra CSS class you want to add to the <CardBody>.
   * Useful for styling the component.
   */
  cardBodyClassName?: string;

  /**
   * Optional extra CSS class you want to add to the <CardFooter>.
   * Useful for styling the component.
   */
  cardFooterClassName?: string;
};

/**
 * Card is a content container with an optional header and footer.
 * You should use this component over the reactstrap Card component,
 * because this component has Suspense built in to display a loader
 * and wait for its content to be fully loaded before displaying.
 */
export function Card({
  header,
  footer,
  children,
  className,
  cardHeaderClassName,
  cardBodyClassName,
  cardFooterClassName
}: Props) {
  return (
    <ReactstrapCard className={className}>
      {header ? (
        <CardHeader className={cardHeaderClassName}>
          <Suspense fallback={<Loading />}>{header}</Suspense>
        </CardHeader>
      ) : null}
      <CardBody className={cardBodyClassName}>
        <Suspense fallback={<Loading />}>
          <ErrorBoundary>
            {children}
          </ErrorBoundary>
        </Suspense>
      </CardBody>
      {footer ? (
        <CardFooter className={cardFooterClassName}>
          <Suspense fallback={<Loading />}>{footer}</Suspense>
        </CardFooter>
      ) : null}
    </ReactstrapCard>
  );
}
