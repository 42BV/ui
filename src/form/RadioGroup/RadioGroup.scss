.radio-group > legend {
  font-size: 1rem;
}
