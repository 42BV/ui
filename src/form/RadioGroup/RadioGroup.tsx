import React from 'react';
import { FormGroup, Input, Label } from 'reactstrap';
import { constant, get } from 'lodash';

import withJarb from '../withJarb/withJarb';
import { Color } from '../types';
import { t } from '../../utilities/translation/translation';
import {
  isOptionSelected,
  OptionEnabledCallback,
  OptionEqual,
  OptionForValue,
  OptionsFetcher
} from '../option';
import { doBlur } from '../utils';
import Loading from '../../core/Loading/Loading';
import { useOptions } from '../useOptions';

export interface Text {
  /**
   * The message to show when the select is loading. Defaults
   * to `loading...`
   */
  loadingMessage?: string;
}

interface BaseProps<T> {
  /**
   * The id of the form element.
   */
  id?: string;

  /**
   * The value that the form element currently has.
   */
  value?: T;

  /**
   * Callback for when the form element changes.
   */
  onChange: (value: T) => void;

  /**
   * Optional callback for when the form element is blurred.
   */
  onBlur?: () => void;

  /**
   * Is either an array of options or a callback which fetches
   * the options asynchronously.
   */
  options: OptionsFetcher<T> | T[];

  /**
   * Callback to convert an value of type T to an option to show
   * to the user.
   */
  optionForValue: OptionForValue<T>;

  /**
   * Optional callback which is used to determine if two options
   * of type T are equal.
   *
   * When `isOptionEqual` is not defined the outcome of `optionForValue`
   * is used to test equality.
   */
  isOptionEqual?: OptionEqual<T>;

  /**
   * Optional callback which is called for every option to determine
   * if the option can be selected. By default all options can be
   * selected.
   */
  isOptionEnabled?: OptionEnabledCallback<T>;

  /**
   * Optionally the error message to render.
   */
  error?: React.ReactNode;

  /**
   * Optionally the color of the FormGroup.
   */
  color?: Color;

  /**
   * Whether or not the form element is currently valid.
   */
  valid?: boolean;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;

  /**
   * The placeholder of the form element.
   */
  placeholder?: string;

  /**
   * Optionally customized text within the component.
   * This text should already be translated.
   */
  text?: Text;
}

interface WithoutLabel<T> extends BaseProps<T> {
  label?: never;
}

interface WithLabel<T> extends BaseProps<T> {
  /**
   * The label of the form element.
   */
  label: string;
}

export type Props<T> = WithoutLabel<T> | WithLabel<T>;

/**
 * RadioGroup is a form element for which the value can be selected
 * from a limited range.
 */
export default function RadioGroup<T>(props: Props<T>) {
  const {
    value,
    error,
    color,
    text = {},
    className = '',
    placeholder,
    onChange,
    onBlur,
    optionForValue,
    isOptionEqual
  } = props;

  const { options, loading } = useOptions({
    optionsOrFetcher: props.options,
    value,
    isOptionEqual,
    optionForValue
  });

  const isOptionEnabled = get(props, 'isOptionEnabled', constant(true));

  function onRadioClicked(option: T) {
    onChange(option);
    doBlur(onBlur);
  }

  return (
    <FormGroup tag="fieldset" className={className} color={color}>
      {'label' in props && props.label ? <legend>{props.label}</legend> : null}
      {placeholder ? (
        <p className="text-muted">
          <em>{placeholder}</em>
        </p>
      ) : null}
      {loading ? (
        <Loading>
          {t({
            key: 'RadioGroup.LOADING',
            fallback: 'Loading...',
            overrideText: text.loadingMessage
          })}
        </Loading>
      ) : (
        options.map(option => {
          const label = optionForValue(option);

          return (
            <FormGroup key={label} check>
              <Label check>
                <Input
                  type="radio"
                  value={label}
                  checked={isOptionSelected({
                    option,
                    optionForValue,
                    isOptionEqual,
                    value
                  })}
                  disabled={!isOptionEnabled(option)}
                  onChange={() => onRadioClicked(option)}
                />{' '}
                {label}
              </Label>
            </FormGroup>
          );
        })
      )}

      {error}
    </FormGroup>
  );
}
/**
 * Variant of the RadioGroup which can be used in a Jarb context.
 */
export const JarbRadioGroup = withJarb<any, any, Props<any>>(RadioGroup);
