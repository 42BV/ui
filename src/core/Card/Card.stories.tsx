import React from 'react';
import { CardTitle } from 'reactstrap';

import { Card } from './Card';

export default {
  title: 'core/Card',

  parameters: {
    component: Card
  }
};

const BasicStory = () => {
  return (
    <Card header={<CardTitle>This is a card</CardTitle>} footer="And a footer">
      <p>With some content in it</p>
    </Card>
  );
};

export const Basic = {
  render: BasicStory,
  name: 'basic'
};

const WithoutHeaderStory = () => {
  return (
    <Card footer="And a footer">
      <CardTitle>This is a card</CardTitle>
      <p>With some content in it</p>
    </Card>
  );
};

export const WithoutHeader = {
  render: WithoutHeaderStory,
  name: 'without header'
};

const WithoutFooterStory = () => {
  return (
    <Card header={<CardTitle>This is a card</CardTitle>}>
      <p>With some content in it</p>
    </Card>
  );
};

export const WithoutFooter = {
  render: WithoutFooterStory,
  name: 'without footer'
};
