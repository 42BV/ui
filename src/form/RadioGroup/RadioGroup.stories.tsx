import React, { useState } from 'react';

import { FieldRadioGroup, JarbRadioGroup, RadioGroup } from './RadioGroup';
import {
  FieldFormElementDependencies,
  FinalForm,
  IsOptionEqualInfo,
  JarbFormElementDependencies,
  KeyForOptionInfo,
  nonExistingProvince,
  Province,
  provinceFetcher,
  provinces,
  ReloadOptionsInfo,
  resolveAfter
} from '../../story-utils';
import { Card } from '../../core/Card/Card';
import { pageOf } from '../../utilities/page/page';
import { Tooltip } from '../../core/Tooltip/Tooltip';
import { Icon } from '../../core/Icon';

export default {
  title: 'Form/RadioGroup',

  parameters: {
    component: RadioGroup
  }
};

const PredefinedOptionsStory = () => {
  const [value, setValue] = useState<Province | undefined>(
    nonExistingProvince()
  );

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}
    </Card>
  );
};

export const PredefinedOptions = {
  render: PredefinedOptionsStory,
  name: 'predefined options'
};

const AsyncOptionsStory = () => {
  const [value, setValue] = useState<Province | undefined>(provinces()[0]);

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinceFetcher}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}

      <p className="mt-3">
        <strong>Disclaimer:</strong> when using async, a maximum of 100 options
        will be displayed. If you want to display more than 100 options, you
        should use the ModalPickerSingle.
      </p>
    </Card>
  );
};

export const AsyncOptions = {
  render: AsyncOptionsStory,
  name: 'async options'
};

const DisabledOptionsStory = () => {
  const [value, setValue] = useState<Province | undefined>(
    nonExistingProvince()
  );

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        isOptionEnabled={(option) => !option.label.startsWith('Z')}
        value={value}
        onChange={setValue}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}
    </Card>
  );
};

export const DisabledOptions = {
  render: DisabledOptionsStory,
  name: 'disabled options'
};

const CustomIsOptionEqualStory = () => {
  const [value, setValue] = useState<Province | undefined>(provinces()[0]);

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        isOptionEqual={(a, b) => a.value === b.value}
        value={value}
        onChange={setValue}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}

      <IsOptionEqualInfo />
    </Card>
  );
};

export const CustomIsOptionEqual = {
  render: CustomIsOptionEqualStory,
  name: 'custom isOptionEqual'
};

const CustomKeyForOptionStory = () => {
  const [value, setValue] = useState<Province | undefined>(provinces()[0]);

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        keyForOption={(province) => province.value}
        value={value}
        onChange={setValue}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}

      <KeyForOptionInfo />
    </Card>
  );
};

export const CustomKeyForOption = {
  render: CustomKeyForOptionStory,
  name: 'custom keyForOption'
};

const UsingReloadOptionsStory = () => {
  const [brand, setBrand] = useState<string>();
  const [model, setModel] = useState<string>();

  const allOptions = {
    Audi: ['A1', 'A2', 'A3', 'M5'],
    BMW: ['series 1', 'series 2', 'series 3', 'series 4', 'series 5'],
    Mercedes: ['Viano', 'Vito', 'Sprinter']
  };

  return (
    <Card className="m-2">
      <RadioGroup
        id="brand"
        label="Brand"
        placeholder="Please select your brand"
        options={() => resolveAfter(pageOf(Object.keys(allOptions), 1))}
        labelForOption={(option) => option}
        value={brand}
        onChange={(value) => {
          setBrand(value);
          setModel(undefined);
        }}
      />
      <RadioGroup
        id="model"
        label="Model"
        placeholder={
          brand ? 'Please select your model' : 'Please select a brand first'
        }
        options={() => resolveAfter(pageOf(brand ? allOptions[brand] : [], 1))}
        labelForOption={(option: string) => option}
        value={model}
        onChange={setModel}
        reloadOptions={brand}
      />

      {brand ? <p>Your chosen brand is: {brand}</p> : null}
      {model ? <p>Your chosen model is: {model}</p> : null}

      <ReloadOptionsInfo />
    </Card>
  );
};

export const UsingReloadOptions = {
  render: UsingReloadOptionsStory,
  name: 'using reloadOptions'
};

const LabelPlaceholderStory = () => {
  const [value, setValue] = useState<Province | undefined>(
    nonExistingProvince()
  );

  return (
    <Card className="m-2">
      <span className="d-block fs-5">Invisible label</span>

      <RadioGroup<Province>
        label="Province"
        hiddenLabel={true}
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
      />

      <hr />

      <span className="d-block fs-5">Custom label</span>

      <RadioGroup<Province>
        id="provinces"
        label={
          <div className="d-flex justify-content-between">
            <span>Subject</span>
            <Tooltip className="ms-1" content="The province you live in">
              <Icon icon="info" />
            </Tooltip>
          </div>
        }
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
      />

      <hr />

      <span className="d-block fs-5">Without placeholder</span>

      <RadioGroup<Province>
        id="province"
        label="Province"
        options={provinces()}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
      />

      <hr />

      {value ? <p>Your chosen province is: {value.label}</p> : null}
    </Card>
  );
};

export const LabelPlaceholder = {
  render: LabelPlaceholderStory,
  name: 'label & placeholder'
};

const HorizontalStory = () => {
  const [value, setValue] = useState<Province | undefined>(
    nonExistingProvince()
  );

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
        horizontal={true}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}

      <p>
        <strong>Disclaimer:</strong> horizontal mode works best when there are
        not too many items
      </p>
    </Card>
  );
};

export const Horizontal = {
  render: HorizontalStory,
  name: 'horizontal'
};

const HorizontalWithClearStory = () => {
  const [value, setValue] = useState<Province | undefined>(
    nonExistingProvince()
  );

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
        horizontal={true}
        canClear={true}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}

      <p>
        <strong>Disclaimer:</strong> horizontal mode works best when there are
        not too many items
      </p>
    </Card>
  );
};

export const HorizontalWithClear = {
  render: HorizontalWithClearStory,
  name: 'horizontal with clear'
};

const WithClearButtonStory = () => {
  const [value, setValue] = useState<Province | undefined>(
    nonExistingProvince()
  );

  return (
    <Card className="m-2">
      <RadioGroup<Province>
        id="province"
        label="Province"
        placeholder="Please select your province"
        options={provinces()}
        labelForOption={(province) => province.label}
        value={value}
        onChange={setValue}
        canClear={true}
      />

      {value ? <p>Your chosen province is: {value.label}</p> : null}
    </Card>
  );
};

export const WithClearButton = {
  render: WithClearButtonStory,
  name: 'with clear button'
};

const FieldStory = () => {
  return (
    <>
      <FieldFormElementDependencies />
      <FinalForm>
        <FieldRadioGroup
          id="province"
          name="province"
          label="Province"
          placeholder="Please select your province"
          options={provinceFetcher}
          labelForOption={(province) => province.label}
        />
      </FinalForm>
    </>
  );
};

export const Field = {
  render: FieldStory,
  name: 'field'
};

const JarbStory = () => {
  return (
    <>
      <JarbFormElementDependencies />
      <FinalForm>
        <JarbRadioGroup
          id="province"
          name="province"
          placeholder="Please select your province"
          options={provinceFetcher}
          labelForOption={(province) => province.label}
          jarb={{
            validator: 'User.province',
            label: 'Province'
          }}
        />
      </FinalForm>
    </>
  );
};

export const Jarb = {
  render: JarbStory,
  name: 'jarb'
};
