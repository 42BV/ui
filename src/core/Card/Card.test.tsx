import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RadioGroup from '../../form/RadioGroup/RadioGroup';

import { Card } from './Card';

describe('Component: Card', () => {
  function setup({
    hasHeader = true,
    hasFooter = true
  }: {
    hasHeader?: boolean;
    hasFooter?: boolean;
  }) {
    const props = {
      header: hasHeader ? 'A simple header' : undefined,
      footer: hasFooter ? 'A simple footer' : undefined
    };

    const children = (
      <RadioGroup<string>
        onChange={jest.fn()}
        options={['local', 'development', 'test', 'acceptation', 'production']}
        labelForOption={(v) => v}
      />
    );

    const card = shallow(<Card {...props}>{children}</Card>);

    return { card };
  }

  describe('ui', () => {
    test('basic', () => {
      const { card } = setup({});

      expect(toJson(card)).toMatchSnapshot('Component: Card => ui => basic');
    });

    test('without header', () => {
      const { card } = setup({ hasHeader: false });

      expect(toJson(card)).toMatchSnapshot(
        'Component: Card => ui => without header'
      );
    });

    test('without footer', () => {
      const { card } = setup({ hasFooter: false });

      expect(toJson(card)).toMatchSnapshot(
        'Component: Card => ui => without footer'
      );
    });
  });
});
