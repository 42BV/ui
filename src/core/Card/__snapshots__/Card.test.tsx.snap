// Vitest Snapshot v1

exports[`Component: Card > ui > basic > Component: Card => ui => basic 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-header card-header"
    >
      A simple header
    </div>
    <div
      class="card-body card-body"
    >
      <fieldset
        class="radio-group  mb-3"
      >
        <legend>
          Environment
        </legend>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="local"
              type="radio"
              value="local"
            />
             
            local
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="development"
              type="radio"
              value="development"
            />
             
            development
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="test"
              type="radio"
              value="test"
            />
             
            test
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="acceptation"
              type="radio"
              value="acceptation"
            />
             
            acceptation
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="production"
              type="radio"
              value="production"
            />
             
            production
          </label>
        </div>
      </fieldset>
    </div>
    <div
      class="card-footer card-footer"
    >
      A simple footer
    </div>
  </div>
</div>
`;

exports[`Component: Card > ui > without footer 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-header card-header"
    >
      A simple header
    </div>
    <div
      class="card-body card-body"
    >
      <fieldset
        class="radio-group  mb-3"
      >
        <legend>
          Environment
        </legend>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="local"
              type="radio"
              value="local"
            />
             
            local
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="development"
              type="radio"
              value="development"
            />
             
            development
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="test"
              type="radio"
              value="test"
            />
             
            test
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="acceptation"
              type="radio"
              value="acceptation"
            />
             
            acceptation
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="production"
              type="radio"
              value="production"
            />
             
            production
          </label>
        </div>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`Component: Card > ui > without header 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-body card-body"
    >
      <fieldset
        class="radio-group  mb-3"
      >
        <legend>
          Environment
        </legend>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="local"
              type="radio"
              value="local"
            />
             
            local
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="development"
              type="radio"
              value="development"
            />
             
            development
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="test"
              type="radio"
              value="test"
            />
             
            test
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="acceptation"
              type="radio"
              value="acceptation"
            />
             
            acceptation
          </label>
        </div>
        <div
          class="form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              defaultvalue="production"
              type="radio"
              value="production"
            />
             
            production
          </label>
        </div>
      </fieldset>
    </div>
    <div
      class="card-footer card-footer"
    >
      A simple footer
    </div>
  </div>
</div>
`;
